### PR TITLE
CSV Import

### PR DESCRIPTION
**Pull request by edwardstamps**
Working Version of CSV Import for multiple content types

### CODE REVIEW

File: web/admin/src/App.js [hunk 1]
```diff
@@ -266,39 +267,74 @@ class App extends PureComponent {
...
+          let newOrderData = item.rawData.slice()
+          newOrderData.forEach(newItem => {
+            newItem.order = item.order
+          })
+          csvData = csvData.concat(newOrderData)
+        }
+      })
+    }
+
+      // 2a. Public bucket gets copies of global content and those with attendee group filters.
```
Comment (AdamLiechty) [inline web/admin/src/App.js]: looks like the indentation got messed up here

File: web/admin/src/ContentDetailsEditor.js [hunk 2]
```diff
@@ -61,6 +135,60 @@ export default class ContentDetailsEditor extends PureComponent {
     }
   }
 
+  isCSV = () => {
+    if (this.props.content.type === "textCSV" || this.props.content.type === "webCSV" || this.props.content.type === "videoCSV") {
```
Comment (AdamLiechty) [inline web/admin/src/ContentDetailsEditor.js]: This function can simply be
```
return ['textCSV', 'webCSV', 'videoCSV'].includes(this.props.content.type)
```

Any time you find yourself writing something like
```
if (a || b && c) {
  return true
} else {
  return false
}
```

remember that `a || b && c` is itself a boolean-typed expression, so you can just say `return a || b && c`

File: web/admin/src/App.js [hunk 3]
```diff
@@ -266,29 +267,54 @@ class App extends PureComponent {
...
     this.usersRef().remove()
     this.tiersRef().remove()
 
     // 2. Create derived copies from `publishedContent`
     const { key, ...contentToPublish } = content
+    let csvData = []
+
+    //check if new object is for csv
+    if (content.rawData) {
+      const publishData = content.rawData.slice()
```
Comment (AdamLiechty) [inline web/admin/src/App.js]: why `slice()`?

File: web/admin/src/App.js [hunk 4]
```diff
@@ -266,29 +267,54 @@ class App extends PureComponent {
...
     const publishedContent = Object.keys({...this.state.publishedContent, [key]: content})
     .map(k => k === key ? content : {...this.state.publishedContent[k], key: k})
     .filter(x => Object.keys(x).length > 1) // Ignore key-only objects that are being unpublished
-    
+
+    //check for reordering thus content blank
+    if (Object.keys(content).length === 0) {
+      publishedContent.forEach(item => {
+        if (item.type === "textCSV" || item.type === "webCSV" || item.type === "videoCSV") {
+          let newOrderData = item.rawData.slice()
```
Comment (AdamLiechty) [inline web/admin/src/App.js]: Why `slice()`?  Is it necessary?  It just creates a new array that has all the same elements.

Comment (edwardstamps) [reply web/admin/src/App.js]: I'm assuming its either legacy code or ensuring a object to not change the underlying info. Let me change to object.assign

Comment (edwardstamps) [reply web/admin/src/App.js]: Yes this is to make a new array without changing the underlying value

**Pull request merged**